### PR TITLE
Zephyr io

### PR DESCRIPTION
**Pull request by edersondisouza**
Some fixes.
@glima please check spi patch - I'm not sure what is wrong, comment or code. I changed code as comment appears correct.

### CODE REVIEW

File: src/lib/io/sol-aio-impl-zephyr.c [hunk 1]
```diff
@@ -138,8 +138,11 @@ static bool
 aio_read_timeout_cb(void *data)
 {
     struct sol_aio *aio = data;
+    int r;
 
-    aio->async.value = adc_read(aio->dev, &aio->table);
+    r = adc_read(aio->dev, &aio->table);
```
Comment (glima) [inline src/lib/io/sol-aio-impl-zephyr.c]: Indeed, my bad here. Thanks.

File: src/lib/io/sol-aio-impl-zephyr.c [hunk 2]
```diff
@@ -96,7 +96,7 @@ sol_aio_open_raw(const int device, const int pin, const unsigned int precision)
     aio = calloc(1, sizeof(struct sol_aio));
     SOL_NULL_CHECK(aio, NULL);
 
-    aio->sample.sampling_delay = 1;
+    aio->sample.sampling_delay = 12;
```
Comment (glima) [inline src/lib/io/sol-aio-impl-zephyr.c]: I also had little idea of what to put here, 1 was sort of a ``FIXME``.

**Comment by glima** [comment]
LGTM

**Comment by glima** [comment]
@edersondisouza, other than that, it was just a matter of flashing to Soletta to arc and changing the device names, when applicable?

**Comment by edersondisouza** [comment]
On 07-04-2016 15:55, Gustavo Lima Chaves wrote:
> @edersondisouza <https://github.com/edersondisouza>, other than that, it
> was just a matter of flashing to Soletta to arc and changing the device
> names, when applicable?

No need to change device name on Soletta side. However, it's important 
to notice options that needed to be set on prj.conf.
I didn't double checked, but my test app final prj.conf looks like:

```
CONFIG_NEWLIB_LIBC=y
CONFIG_STDOUT_CONSOLE=y
CONFIG_MICROKERNEL=n
CONFIG_NANO_TIMEOUTS=y

CONFIG_ADC=y
CONFIG_ADC_DW=y
CONFIG_ADC_DW_REPETITIVE=y

CONFIG_ADC_DW_SAMPLE_WIDTH=3

CONFIG_SPI=y
CONFIG_SPI_DW_INTERRUPT_SEPARATED_LINES=y

CONFIG_I2C=y
CONFIG_I2C_QUARK_SE_SS_1=y
```

**Comment by ibriano** [comment]
On Thu, 07 Apr 2016 11:58:27 -0700, Ederson de Souza wrote:
> On 07-04-2016 15:55, Gustavo Lima Chaves wrote:
> > @edersondisouza <https://github.com/edersondisouza>, other than that, it
> > was just a matter of flashing to Soletta to arc and changing the device
> > names, when applicable?
> 
> No need to change device name on Soletta side. However, it's important 
> to notice options that needed to be set on prj.conf.
> I didn't double checked, but my test app final prj.conf looks like:
> 

[ ... ]

> CONFIG_MICROKERNEL=n

Redundant, microkernel is always disabled for ARC, and in any case, that
should be passed to make as KERNEL_TYPE.

> CONFIG_ADC_DW_REPETITIVE=y
> CONFIG_ADC_DW_SAMPLE_WIDTH=3

Why these two options? If they are needed for the driver to work, they
should have been set by default.

> CONFIG_SPI_DW_INTERRUPT_SEPARATED_LINES=y

Same thing applies here.

**Comment by edersondisouza** [comment]
On 07-04-2016 16:05, Iván Briano wrote:
> On Thu, 07 Apr 2016 11:58:27 -0700, Ederson de Souza wrote:
>  > On 07-04-2016 15:55, Gustavo Lima Chaves wrote:
>  > > @edersondisouza <https://github.com/edersondisouza>, other than
> that, it
>  > > was just a matter of flashing to Soletta to arc and changing the device
>  > > names, when applicable?
>  >
>  > No need to change device name on Soletta side. However, it's important
>  > to notice options that needed to be set on prj.conf.
>  > I didn't double checked, but my test app final prj.conf looks like:
>  >
>
> [ ... ]
>
>  > CONFIG_MICROKERNEL=n
>
> Redundant, microkernel is always disabled for ARC, and in any case, that
> should be passed to make as KERNEL_TYPE.
>
>  > CONFIG_ADC_DW_REPETITIVE=y
>  > CONFIG_ADC_DW_SAMPLE_WIDTH=3

SAMPLE_WIDTH depends more on the sensor one is using than on the driver. 
REPETITIVE is an option to INTERRUPTION - I may check if it works 
without it.

> Why these two options? If they are needed for the driver to work, they
> should have been set by default.
>
>  > CONFIG_SPI_DW_INTERRUPT_SEPARATED_LINES=y

Well, it failed to compile without it - guess that zephyr defaults are 
messed up.

**Comment by edersondisouza** [comment]
On 07-04-2016 16:05, Iván Briano wrote:
> On Thu, 07 Apr 2016 11:58:27 -0700, Ederson de Souza wrote:
>  > On 07-04-2016 15:55, Gustavo Lima Chaves wrote:
>  > > @edersondisouza <https://github.com/edersondisouza>, other than
> that, it
>  > > was just a matter of flashing to Soletta to arc and changing the device
>  > > names, when applicable?
>  >
>  > No need to change device name on Soletta side. However, it's important
>  > to notice options that needed to be set on prj.conf.
>  > I didn't double checked, but my test app final prj.conf looks like:
>  >
>
> [ ... ]
>
>  > CONFIG_MICROKERNEL=n

Not sure now, but I vaguely remember that Soletta was believing that it 
was on microkernel without this - but this was when I was trying 
everything while stuck on adc...

**Comment by edersondisouza** [comment]
Integrated. Tks!

**Pull request closed without merging**